frontend/src/helpers/dropdownMappers.ts

```diff
@@ -121,6 +121,16 @@ export const processFromEntitiesData = (allOptions: tempOption[]) => {
 export function getViasGroupLabel(currentIndex: number | null) {
   return currentIndex ? `${ViasGroupLabel}-${currentIndex}` : OriginsGroupLabel;
 }
+export function areArraysOfObjectsEqual(a: any[], b: any[]): boolean {
+  if (a.length !== b.length) return false;
+  
+  const setA = new Set(a.map(obj => obj.id));
+  const setB = new Set(b.map(obj => obj.id));
+  
+  return JSON.stringify([...setA].sort()) === JSON.stringify([...setB].sort());
```

Comment:
I think we can avoid using JSON.stringify for comparation and instead use something like:
```
export function areArraysOfObjectsEqual(a: any[], b: any[]): boolean {
  if (a.length !== b.length) return false;

  const setA = new Set(a.map(obj => obj.id));
  const setB = new Set(b.map(obj => obj.id));

  if (setA.size !== setB.size) return false;

  for (let id of setA) {
    if (!setB.has(id)) return false;
  }

  return true;
}
```

The idea behind creating a Set is so that we can perform a check if an element is present in constant time O(1), without the need to sort (which would take more time, O(n log n))

Reply:
@afonsobspinto what is there is an id in setB that is not in setA

Reply:
I think you might have missed this line @zsinnema :
```
  if (setA.size !== setB.size) return false;
```

If the sets have different sizes for sure they are not equal.
If the sets have equal size, it doesn't matter if all of B are in A or if all of A are in B.

Reply:
@afonsobspinto agree :-)